### DISCUSSION TITLE
Downloader randomly pauses and wont resume. Unable to diagnose with -v

### DISCUSSION THREAD

**Discussion by ScootyBooty** [Serene-Arc/bulk-downloader-for-reddit#454] [Q&A]
So I'm using this downloader to get all media from a user's reddit account with the following command

python -m bdfr download C:\BDFR --submitted --user jessnjamez --file-scheme {POSTID} --max-wait-time 10 --folder-scheme {REDDITOR} --no-dupes -v

(this is a NSFW account)

for most users, this executes just fine. However, for some users, the downloader reaches a point at which it just pauses and never resumes. This is shown below. 
![image](https://user-images.githubusercontent.com/28027267/121398294-34b13880-c923-11eb-9d6e-10453d9319e1.png)

it has currently been ~8 minutes since it stopped. 
![image](https://user-images.githubusercontent.com/28027267/121398375-472b7200-c923-11eb-8d06-342cef4a2c84.png)

Any thoughts?

**Comment by Serene-Arc** [comment]
It is probably just downloading. The resource it is stopped on is a 20 minute video, if your internet is slow then it may take some time to download everything. Leave it for longer and see if it completes.

**Comment by LGSarge87** [comment]
So I'm seeing this also, will just stop on particular submissions by certain users. There's no real trend though, sometimes a reddit hosted image, sometimes Imgur, sometimes an external link. I'm on gigabit, so its not a connection speed issue or a large download (I checked network traffic too, it stopped).

I'm pretty sure it was introduced with v 2.2.0, as this wasnt a problem prior to that.

**Reply by Serene-Arc** [reply]
This isn't enough information to diagnose the problem. What submissions are they? What stage does the downloader stop working? What's the last logging message? You having gigabit internet isn't necessarily helping, if it is hanging on a network call. Too many calls in a short time can make some websites simply drop connections silently, or do other things that aren't technically to spec.

**Reply by LGSarge87** [reply]
Completely understand, was just wanting to add to the discussion that this isn't an isolated issue. I'll send through what I can when I gather the details to help figure out what's going wrong.

I don't think it's an issue with dropping connections, as it's exactly the same submissions each time that causes the issue (and these links are sometimes imgur, sometimes reddit, sometimes external).

**Reply by Serene-Arc** [reply]
It's possible our rate-limiting code is malfunctioning I suppose, or that praw is. I'd need the data though, open an issue when you've got it :)

**Comment by RealAwkwardPig** [comment]


**Reply by Serene-Arc** [reply]
Why do you think that this is the problem? 5 seconds is way too short to show anything. Do you *know* that part of the code is causing the problem or are you assuming it? Because the problem seems to have been narrowed down to PornHub videos which don't have anything to do with the `requests` module. It's not even used.
